README.md

```diff
@@ -71,3 +62,44 @@ services:
       - postgresql:/var/lib/postgresql
       - postgresql_data:/var/lib/postgresql/data
 ``` 
+
+# Configuration
+## Global configuration
+If you can define the properties globally (e.g. the pull request provider) within your SonarQube instance go to [/admin/settings?category=pull+request](http://localhost:9000/admin/settings?category=pull+request) and set
+the properties as admin.
+
+Set all other properties that you can define globally for all of your projects.
+
+Make sure `sonar.core.serverBaseURL` in SoanrQube [/admin/settings](http://localhost:9000/admin/settings) is properly set in order to for the links in the
+ comment to
+ work.
+
+## Project configuration
+Override global configuration on project level at [/project/settings?category=pull+request&id=PROJECT_KEY](http://localhost:9000/project/settings?category=pull+request&id=PROJECT_KEY) and set project specific
+ settings here.
+
+# Configuration of Bitbucket Server
+Set properties for Bitbucket Server either on Global configuration or Project configuration level as mentioned in the
+ Configuration section.
+
+## How to decorate the PR
+### Run analysis of target branch
+In order to decorate your short living branches (hotfixes, features, ...) you need to analyze your long living
+ branches (master, development, release branches) first. A configuration what long living branches are can be also
+  done in SonarQube.
+  
+The analysis needs the following setting:
+`sonar.branch.name = master`
+
+### Run analysis of the PR branch
+Carefully read the official SonarQube guide for [pull request decoration](https://docs.sonarqube.org/latest/analysis/pull-request/) 
+
+In there you'll find the following properties that need to be set.
+```
+sonar.pullrequest.key = pull_request_id (e.g. 100)
+sonar.pullrequest.branch = source_branch_name (e.g feature/TICKET-123)
+sonar.pullrequest.base = target_branch_name (e.g master)
+```
+
```

Comment:
I'd possibly also mention to ensure you're not including any `sonar.branch` properties otherwise your Pull Request will be treated as a branch and wont be decorated

Reply:
very good input!